Create CONTRIBUTING.md

Covkie opened the pull request:
Not finished needs discussion.

@kirikarma Need the Image for Art contribution
-Art and Language Snippets will be added to the Assets repo (soon™)

@skyrina What will the subdomain for our localization solution be? At the moment i have the link pointing to: https://localization.figuramc.org

@UnlikePaladin, @omoflop, @ThePlasticPotato Anything that I missed?

Covkie committed:
Create CONTRIBUTING.md

Not finished needs discussion.

lenrik1589 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,117 @@
+## Pull Request Process
+note - do not merge upstream into your pr branch if a clean rebase is possible
+
+1. Ensure the feature is In demand via the `📧丨mod-suggestions` channel in our [Discord](https://discord.gg/figuramc) or has the [Enhancment]([https://github.com/FiguraMC/Figura/labels/enhancement](https://github.com/FiguraMC/Figura/issues?q=is%3Aissue+is%3Aopen+label%3Aenhancement+)) tag in the Issues tab 
```
If someone is implementing a feature, it is likely it is in demand at least with them, and they can add a suggestion before making a PR so this requirement seems kinda pointless.

aprilkenopsia replied:
```suggestion
1. Ensure the feature has not been denied and is in demand via the `📧丨mod-suggestions` channel in our [Discord](https://discord.gg/figuramc) or has the [Enhancment]([https://github.com/FiguraMC/Figura/labels/enhancement](https://github.com/FiguraMC/Figura/issues?q=is%3Aissue+is%3Aopen+label%3Aenhancement+)) tag in the Issues tab.
```

lenrik1589 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,117 @@
+   
+    OR:
+
+   Ensure what you are doing is actually a bug via the `👾丨bug-reports` channel in our [Discord](https://discord.gg/figuramc) or has the [Bug]([https://github.com/FiguraMC/Figura/issues](https://github.com/FiguraMC/Figura/issues?q=is%3Aissue+is%3Aopen+label%3Abug+)) tag in the Issues tab 
```
Same here, although confirming that something *is* a bug is more important as there are intentional limitations in multiple places.

aprilkenopsia replied:
```suggestion
   Ensure what you are doing is actually a bug via the `👾丨bug-reports` channel in our [Discord](https://discord.gg/figuramc) or has the [Bug]([https://github.com/FiguraMC/Figura/issues](https://github.com/FiguraMC/Figura/issues?q=is%3Aissue+is%3Aopen+label%3Abug+)) tag in the Issues tab.
```

lenrik1589 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,117 @@
+
+4. Clearly and extensively document everything you've added/changed in the pull request. This helps when compiling all the changes into full release patch notes.
+
+5. You may merge the Pull Request in once you have the sign-off of another developer, or if you do not have permission to do that, you may request the second reviewer to merge it for you.
```
You may *not* merge Pull Request, without being a collaborator on the repository, "Reviewer" does not grant write access either. "Your pull request will be merged by one of the developers on their own discretion, constantly asking developers to accept your Pull Request *might* work, but is more likely to just annoy them and postpone the merge."

Covkie replied:
typically the people who are reviewing the PR is someone who can merge it. As it says if you can not you can request someone with the correct permissions to do so

lenrik1589 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,117 @@
+
+   Ensure what you are doing is actually a bug via the `👾丨bug-reports` channel in our [Discord](https://discord.gg/figuramc) or has the [Bug]([https://github.com/FiguraMC/Figura/issues](https://github.com/FiguraMC/Figura/issues?q=is%3Aissue+is%3Aopen+label%3Abug+)) tag in the Issues tab 
+
+3. Communicate with Devs and other contributors about what you are building and adding. This helps keep multiple people from working on the same thing alone.
```
That is good, now that there are more than 2 devs working on the front-end it should help

lenrik1589 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,117 @@
+
+3. Communicate with Devs and other contributors about what you are building and adding. This helps keep multiple people from working on the same thing alone.
+
+4. Clearly and extensively document everything you've added/changed in the pull request. This helps when compiling all the changes into full release patch notes.
```
I think "Provide a clear list of changes, fixes, and potential/actual regressions introduced by your PR to help with making change log for the next release" would be better, as currently it is implied that almost every line needs to have a comment (which is, while helpful to people just studying off the Figura codebase (which I doubt is many because of complexity of the mod), may not be that beneficial to other devs and creates unnecessary code clutter).

Covkie replied:
Your suggested replacement says essentially the same thing... I don't see the point of this comment

lenrik1589 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,117 @@
+
+## Translation Contributing Process 
+note - Translations are not located in this repository but in [Figuramc/Assets](https://github.com/FiguraMC/Assets)
+1. You must be fluent and/or competent in the language you are translating to.
```
Assets repo is still private :skull:
But, I don't think "competent" is applicable to knowledge of a language, just "fluent" would be enough.

aprilkenopsia replied:
```suggestion
1. You must be fluent and/or knowledgeable in the language you are translating to.
```

lenrik1589 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,117 @@
+note - Translations are not located in this repository but in [Figuramc/Assets](https://github.com/FiguraMC/Assets)
+1. You must be fluent and/or competent in the language you are translating to.
+
+2. Please use our Localization solution located [here](https://localization.figuramc.org). This makes it much eaiser for other contributors to view the progress on a specific language.
```
Same sentiment, before telling people to use a tool, please publish it first.

Covkie replied:
This is still work in progress and most likely wont be merged before the localization site is setup.

lenrik1589 replied:
That is why I asked to mark it as a draft (and it should probably not be on 0.1.2 "milestone" as that needs to be released soon)

lenrik1589 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,117 @@
+2. Please use our Localization solution located [here](https://localization.figuramc.org). This makes it much eaiser for other contributors to view the progress on a specific language.
+
+## Art Contribution Process
+1. follow the color pallete and style; provided here: [future link here.. sort out before pulling]
```
Yup, this would be better on the point above as well but should be in discussion as well as a part of "open questions".

Covkie replied:
it is a placeholder.

lenrik1589 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,117 @@
+
+4. contact information (optional)
+
+should we implement a solution like: https://github.com/all-contributors/all-contributors ?
```
This should be a discussion, not in the body of the PR as well.

lenrik1589 reviewed (changes requested):
This PR should still be marked as a draft and should contain a mention of the fact that PRs that are still WIP should be marked as draft.

UnlikePaladin commented:
Looks fine/good to me

gloomsys commented:
Line 8, 23: 'note' capitalized
Line 16: Devs -> developers
Line 29: follow -> Follow | removed unnecessary semicolon
Line 31: document -> Document
Line 36: what -> What
Line 38: include -> Include
Line 40: contact -> Contact

Re: Line 42: yes

Line 101: needs an email address
[CONTRIBUTING-REVIEW.md](https://github.com/FiguraMC/Figura/files/12245211/CONTRIBUTING-REVIEW.md)

Covkie commented:
> Line 8, 23: 'note' capitalized
> 
> Line 16: Devs -> developers
> 
> Line 29: follow -> Follow | removed unnecessary semicolon
> 
> Line 31: document -> Document
> 
> Line 36: what -> What
> 
> Line 38: include -> Include
> 
> Line 40: contact -> Contact
> 
> 
> 
> Re: Line 42: yes
> 
> 
> 
> Line 101: needs an email address
> 
> [CONTRIBUTING-REVIEW.md](https://github.com/FiguraMC/Figura/files/12245211/CONTRIBUTING-REVIEW.md)
> 
> 

I'm 90% sure you can suggest changes instead of doing it this way

gloomsys commented:
thats crazy :exploding_head:

aprilkenopsia commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,117 @@
+Please note we have a code of conduct, please follow it in all your interactions with the project.
+
+## Pull Request Process
+note - do not merge upstream into your pr branch if a clean rebase is possible
```
```suggestion
Note - do not merge upstream into your pr branch if a clean rebase is possible.
```

aprilkenopsia replied:
kiri change

aprilkenopsia commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,117 @@
+
+   Ensure what you are doing is actually a bug via the `👾丨bug-reports` channel in our [Discord](https://discord.gg/figuramc) or has the [Bug]([https://github.com/FiguraMC/Figura/issues](https://github.com/FiguraMC/Figura/issues?q=is%3Aissue+is%3Aopen+label%3Abug+)) tag in the Issues tab 
+
+3. Communicate with Devs and other contributors about what you are building and adding. This helps keep multiple people from working on the same thing alone.
```
```suggestion
3. Communicate with developers and other contributors about what you are building and adding. This helps keep multiple people from working on the same thing alone.
```

aprilkenopsia replied:
kiri change

aprilkenopsia commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,117 @@
+5. You may merge the Pull Request in once you have the sign-off of another developer, or if you do not have permission to do that, you may request the second reviewer to merge it for you.
+
+## Translation Contributing Process 
+note - Translations are not located in this repository but in [Figuramc/Assets](https://github.com/FiguraMC/Assets)
```
```suggestion
Note - Translations are not located in this repository but in [Figuramc/Assets](https://github.com/FiguraMC/Assets)
```

aprilkenopsia replied:
kiri change

aprilkenopsia commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,117 @@
+2. Please use our Localization solution located [here](https://localization.figuramc.org). This makes it much eaiser for other contributors to view the progress on a specific language.
+
+## Art Contribution Process
+1. follow the color pallete and style; provided here: [future link here.. sort out before pulling]
```
```suggestion
1. Follow the color pallete and style provided here: [future link here.. sort out before pulling]
```

aprilkenopsia replied:
kiri change

aprilkenopsia commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,117 @@
+## Art Contribution Process
+1. follow the color pallete and style; provided here: [future link here.. sort out before pulling]
+
+2. document the additions and changes made with your contribution.
```
```suggestion
2. Document the additions and changes made with your contribution.
```

aprilkenopsia replied:
kiri change

aprilkenopsia commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,117 @@
+## Credits
+1. We credit our cobtributors for working on our projects, and as such we would like to credit them appropriately
+
+2. what contribution you made, such as a link to a commit, translations, textures, etc.
```
```suggestion
2. What contribution you made, such as a link to a commit, translations, textures, etc.
```

aprilkenopsia replied:
kiri change

aprilkenopsia commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,117 @@
+
+2. what contribution you made, such as a link to a commit, translations, textures, etc.
+
+3. include the name you would like to be used for the credits page (optional)
```
```suggestion
3. Include the name you would like to be used for the credits page. (optional)
```

aprilkenopsia replied:
kiri change

aprilkenopsia commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,117 @@
+
+3. include the name you would like to be used for the credits page (optional)
+
+4. contact information (optional)
```
```suggestion
4. Contact information. (optional)
```

aprilkenopsia replied:
kiri change

aprilkenopsia commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,117 @@
+2. document the additions and changes made with your contribution.
+
+## Credits
+1. We credit our cobtributors for working on our projects, and as such we would like to credit them appropriately
```
```suggestion
1. We credit our contributors for working on our projects, and as such we would like to credit them appropriately.
```

aprilkenopsia replied:
minor spelling mistake

aprilkenopsia reviewed (changes requested):
🥙 kiri changes + other minor things

Covkie committed:
Apply suggestions from code review

@kirikarma changes

Co-authored-by: april kenopsia <altkirikiri@gmail.com>

Covkie committed:
Update CONTRIBUTING.md

Updated contact info

aprilkenopsia commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,117 @@
+5. You may merge the Pull Request in once you have the sign-off of another developer, or if you do not have permission to do that, you may request the second reviewer to merge it for you.
+
+## Translation Contributing Process 
+Note - Translations are not located in this repository but in [Figuramc/Assets](https://github.com/FiguraMC/Assets)
```
```suggestion
Note - Translations are not located in this repository but in [FiguraMC/Assets](https://github.com/FiguraMC/Assets)
```

aprilkenopsia commented:
to do
- have https://localization.figuramc.org set up
- have lang files actually in the [FiguraMC/Assets](https://github.com/FiguraMC/Assets)
- add the color pallete and style guide @kirikarma (i think you are in hiatus tho take ur time
- finish credits section (line 33)

Covkie committed:
Fix capitalization

Co-authored-by: april kenopsia <altkirikiri@gmail.com>

Covkie committed:
Disclaimer and Grammar

-Added disclaimer referencing he assets repo
-fixed a grammar inconsistency
-Added Emoji Note

Covkie committed:
Merge branch 'FiguraMC:1.20' into 1.20

Covkie commented:
> pull/42#issuecomment-1694735901
1. `https://localization.figuramc.org/` is no longer planned as we are using hosted Weblate now. https://hosted.weblate.org/projects/figuramc/
2. Lang files will be kept in this repository (FiguraMC/Figura) to prevent older versions of Figura from breaking. Ex; strings being removed or renamed will cause issues on old versions of the mod.
3. Kiri is finally working on it.
4. I'll get on that tmrw

ElectricDaFearox commented:
Imagine, can't be me.

Covkie committed:
Corrected Translation and Art Info

Corrected Translation and Art Info. Credits are still undecided maybe use a CREDITS.md

Covkie committed:
Add Weblate To README

- Did a few grammar fixes as well as general clean ups.

- the Website modal image is not working and i'm too tired to figure this out. Will fix tmrw

Covkie committed:
Update README.md

fixed website icon

aprilkenopsia commented on README.md:
```diff
@@ -26,23 +26,32 @@
+[collective]: https://img.shields.io/badge/Open%20Collective-83b3fb?logo=opencollective&logoColor=ffffff
 [wiki]: https://img.shields.io/badge/Figura-Wiki-black?logo=data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAADAAAAAwCAYAAABXAvmHAAAAAXNSR0IArs4c6QAAAM5JREFUaIHtmbENAjEMRRPECIiSmvrYAZobggXoGIMhGOIa2IHUNDTU7HAsYBeWgh6W/isTx7knS5Z1qaUTwzTPkfg21trj3kWPJCQSoJEATbgTeN2mbYJ53vZ6tDulr4AEaCRAIwEaCdBIgMadO4aLM/PsnfjxELq4TTc7z92JP9szUvoKSIBGAjQSoJEAjQRoqjfzlK194HPc/fBzSlldH/bG015OXwEJ0EiAZhk94HWJXn+no6SvgARoJ ...
+[weblate]: https://img.shields.io/badge/Weblate-Translating-125252?logo=Weblate&logoColor=ffffff&labelColor=2ECCAA
+[website]: https://img.shields.io/badge/Weblate-purple?logo=data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAANwAAADcCAYAAAAbWs+BAAAA0GVYSWZJSSoACAAAAAoAAAEEAAEAAADcAAAAAQEEAAEAAADcAAAAAgEDAAMAAACGAAAAEgEDAAEAAAABAAAAGgEFAAEAAACMAAAAGwEFAAEAAACUAAAAKAEDAAEAAAACAAAAMQECAA0AAACcAAAAMgECABQAAACqAAAAaYcEAAEAAAC+AAAAAAAAAAgACAAIAEgAAAABAAAASAAAAAEAAABHSU1QIDIuMTAuMzQAADIwMjM6MTA6MDQgMTQ6MjM6MzIAAQABoAMAA ...
```
```suggestion
[website]:   https://img.shields.io/badge/Website-purple?logo=data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAG4AAABuCAYAAADGWyb7AAAACXBIWXMAAAsSAAALEgHS3X78AAAAG3RFWHRTb2Z0d2FyZQBDZWxzeXMgU3R1ZGlvIFRvb2zBp+F8AAAHu0lEQVR4nO3dTWgcZRjA8X9AoR6sFKXoJdQI/SBiTkFQEOvRg6daoQc96EEh0RRRL6VC8CKxIEaoYAO2YEHpQQQLvWgvFmzpoaUVDSg9uWDRmkBBEIwH+7Szs/PxPu/7vO+8u9n/KZuP2YEfz8xOdmZ2YmNjg2HtpZMErfzxA0xYrUvqJoYBLhRI2zCAZguXGquuXBGzgssFC2Dr+cHvLX+YD2IWcLmDlcsBsFO4YQMr1yVgJ3DDDlauC8CkcKMGVi4lYBK4UQcrlwIwOlwuaCnAisXGiwqXA1pqsGIx8aLAbXawcjEAzeG6QptbcPu9EwfirkdVB3sw9YUtnilcCjQBevmht0yWt9JbAuKAHuz1P7bEM4OLiTa3YAfVlgVkGayYFZ4JXAy0lFhNrfSWnBGbwIpZ4AXDWaPlAlauCdAVrFgoXhCcJdrcAjz1117mnjyq+rszf65YrYJTRUAfsGIheN5wVmhzC/Dl9E8Wi0qK+NXuJZPl+OJ5wVmhbf/UBqxcKsAu8dRwuaNJo46ngrNAs9w0tpUKb6W3xP17w5ejwUsKF3vKqhqm/V4UuGFEk0YRzwlumNGkz398t+/xAw9ORnsuH7ype5fvPDg234rXCheKlnKf1lYZr5g1pCteH1ixFryocDmhSU14khViE14tmBQCN2pokgse2ACW8VrBijXgmcNtPQ8vnswXDdzhpBBAOVRQgUk+cFq04jvOWy7liyalxLs8GzC5NXjBcOVTBKzR9l/dXfuzkKnWwkm+gN54GjgXtKpzOqw2kU1YdWmf1xcO8sBTw9WdhGOB5gNWTrMOqfFWekvMPmezrxuAq0NrO2sqdBNpgSa54oXAgR+e1dS1wrmc5hY6bZZoksv6hMKBHs9q6vrgimia8xJDpi0GmtSGZwEHejyLqRuA055IGjJtMdGkpnWzggMdntfU7bgOhxar4eYX9AfcvtOWAk2qw7OEAx2e89TtuH7n6yo4H7Tcp02qW8ePv3+Nbdu2mz2P6dQVwYrdwguC6108zsN3Pa79s6RoUhXeB2f3sOXupwHMAIOnrg5MGsP9Dwd0htc3dW1gUhHOdzP5/q7RgANbPBXc64f1T3BoccIbrnfxOIAargs0qYxXhINu8C4/f49+4WO4PQO/Y4WXJZxsJiUNXi5wVWhSSjyvzaUvnEybNKpwEIY3hjPKFQ5s8KJtLg8tTkzMvaG/eKAMB+54OcC1oUHaqfPZz43hairCQdypU8NN3bSDg7zxNJtJKdXm8vLspNsB+NTN21+q4cqvKIsNA5wrGthMnQlcAUxSwR3swR/f1cOBG95mgoN2vFq4CjDJCa54yWwbHOSH54MGg3AQZ+oG4BrApEa4qmucXeCgHS81nBYNOoBzAJMq4ZouSneFgzzwfNEgMdwz11TL7INzuYuABu6bHz5rvYtC7HNOfNFgCOA0t33QwgGd4IWiQTUc6PGiwP2yX3c44AMHafFiosEmg4M0eBZoMIar/H4MQCswaeTgAN7+LQyuWAiiNZbUhAZjuL7+/uesdjWilT3c9BhuoDY0iAD3zr9uC5q+8+XExsYGv76geyM1Bhx0j+eCBh3ATZcezxyZ8IKzeIFSV1d4XaFBA1wZTMoRDtLjuaJBov1bHZiUKxykw9OgQeTNZBuY5AMnt33Y9/NW1z/JFk+LBpHgXMEkDVz5Ph0auHOrp7lx43fl2v1fLLws0GYnYf6aej1uwwG1eHU3Vrnw9bzz5vLc6mkAbzywA/QBk8zhTjgeBhSbOdJ/tU4Zru1OOBo4CJu6clrEECwpyqtJSzjNrYs0m0uwxUtdTptJKN914ZVl1avLLqcuZblNGwTCweaYutymDQzgYLTxspk2aICDJFMHw4EXBc1g2sAITruvk3LGi4EGNtMGdbc9TDR1kCdeNDSjaQNDOBgNvFhoYDdt0HRr302Ilxva2sxN7uMT3R1i1XC3zn2/8NGi1/4OusXLCW1t5s6p6Ho4cMOruMpk35lHHFavOov/a2qKCQa6/VoRDOrRIASu5UK8EDyIDxgbTHKZtjKY5A8Hg3iuty7CDg/sAFOBQTtaHRg0o4EGTgFWLBQP+gElV0jfixGf2Pls3+PV9Suqv29CawKTwuEA3jsc9Pk6FnhSFaJVMl07tz5a+XNXvDo0FzBoRwPN58dlhCdpEcubvTqgptrwqtBcwcANDbQftZkhXhfV4ZXRNGDgjgaJ4SDsOC+XquCKaFowKR4chOEVrnHed/Qx78V0XRGueJzmCwY6NPD9HHAtXs1F6cOKJ3AyZSFgoEeDgA9wd8JzuYvAtzu8/7/ZRUW0UDDwQ4MQOKjHU9z24XZDALi6foXLs5OsHbtqsjxfNAiFg348H7BymQKe2rVuBgZhaGABB3DyTYOFlMoE0BoMwtHACg7i4EmJEU/tWgcwBwMbNLCEA3u8qoshlncA/m/YViVQFi/rm7JCA2s4sMHTXr1yC9O5mvfHYoGBLRrEgAN/PC2YUTHBwB4NYsFJroBjMHVx4aAZb0TBIC4apICDQbwRBoP4aJAKTroU8ZChoVECk9LCSYkARxFM6gZOigQ4ymBSt3CSEeBmAJPygJM8ATcTmJQXXDEHxFF4We9bvnDFSojDfOBs1XDA1bTGq0ErPwxAdf0Hwx3OLyPuOrkAAAAASUVORK5CYII=
```

smoller

Covkie committed:
Update README.md

real work ( i swear)

Co-authored-by: april kenopsia <altkirikiri@gmail.com>

Covkie committed:
Real fix to  figuramc website badge

Okay actually fixed now

Covkie committed:
Rewrite some things in CONTRIBUTING.md

Kiri is busy and is unable to finish the art guidelines so that section is commented out and on hold.

Covkie commented:
All that is left to do is figure out the credits section. Should be ready for merging once that is done.

Covkie committed:
Finish CREDITS and update CONTRIBUTING/README.md

create credits file with all credits - most likely missing a few

update internal repo links to point to HEAD instead of 1.20

UnlikePaladin reviewed (approved):
Looks good to me
